script/tool/lib/src/common/cmake.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:flutter_plugin_tools/src/common/core.dart';
+import 'package:platform/platform.dart';
+
+import 'process_runner.dart';
+
+const String _cmakeCacheCommandKey = 'CMAKE_COMMAND:INTERNAL';
+
+/// A utility class for interacting with CMake projects.
+class CMakeProject {
+  /// Creates an instance that runs commands for [project] with the given
+  /// [processRunner].
+  CMakeProject(
+    this.flutterProject, {
+    required this.buildMode,
+    this.processRunner = const ProcessRunner(),
+    this.platform = const LocalPlatform(),
+  }) {
+    // On Linux 'cmake' is expected to be in the path, so doesn't need to
+    // be lookup up and cached (see [getCMakeCommand]).
+    if (platform.isLinux) {
+      _cmakeCommand = 'cmake';
+    }
+  }
+
+  /// The directory of a Flutter project to run Gradle commands in.
+  final Directory flutterProject;
+
+  /// The [ProcessRunner] used to run commands. Overridable for testing.
+  final ProcessRunner processRunner;
+
+  /// The platform that commands are being run on.
+  final Platform platform;
+
+  /// The build mode (e.g., Debug, Release).
+  ///
+  /// This is a constructor paramater because on Linux many properties depend
+  /// on the build mode since it uses a single-configuration generator.
+  final String buildMode;
+
+  String? _cmakeCommand;
+
+  /// The project's platform directory name.
+  String get _platformDirName => platform.isWindows ? 'windows' : 'linux';
+
+  /// The project's 'example' build directory for this instance's platform.
+  Directory get buildDirectory {
+    Directory buildDir =
+        flutterProject.childDirectory('build').childDirectory(_platformDirName);
+    if (platform.isLinux) {
+      buildDir = buildDir
+          // TODO(stuartmorgan): Support arm64 if that ever becomes a supported
+          // CI configuration for the repository.
+          .childDirectory('x64')
+          // Linux uses a single-config generator, so the base build directory
+          // includes the configuration.
+          .childDirectory(buildMode.toLowerCase());
+    }
+    return buildDir;
+  }
+
+  File get _cmakeCacheFile => buildDirectory.childFile('CMakeCache.txt');
+
+  /// Returns the CMake command to run build commands for this project.
+  ///
+  /// Assumes the project has been built at least once, such that the CMake
+  /// generation step has run.
+  String getCMakeCommand() {
```

Comment:
In dart you should use properties instead of having a `get` method.

Reply:
The Flutter style guide [disagrees](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#getters-feel-faster-than-methods). Loading and parsing a file isn't a trivial operation.

Reply:
Oof seems like a bad choice mixing implementation details with semantics, sgtm.